pkg/universe.dagger.io/x/paul@fgtech.fr/ansible/playbook.cue

```diff
@@ -0,0 +1,110 @@
+package ansible
+
+import (
+	"strings"
+	"dagger.io/dagger"
+	"universe.dagger.io/docker"
+)
+
+#SSHKeys: {
+	// Folder containing ssh keys to mount [required]
+	source: dagger.#FS
+
+	// Folder path where to mount keys in the running container (ie: the one referenced in the Playbook configuration) [default="/root/.ssh"]
+	destPath: string | *"/root/.ssh"
+}
+
+#Inventory: {
+	// Folder where the inventory file is located [required]
+	source: dagger.#FS
+
+	// Name of the inventory file [required]
+	file: string
+}
+
+#PlaybookSource: {
+	// Folder containing the ansible project sources (with tasks, vars, etc.) [required]
+	source: dagger.#FS
+
+	// Main playbook relative path to run in the project directory [default="main.yml"]
+	mainPlaybook: string | *"main.yml"
+
+	// Output path of the playbook generated files [default = "./output"]
+	outputPath: string | *"/output"
+}
+
+#Playbook: {
+	// Ansible Config for running the playbook [required]
+	configImage: docker.#Image
+
+	// Inventory to use for the playbook [required]
+	inventory?: #Inventory
+
+	// Playbook to run [required]
+	playbookSource: #PlaybookSource
+
+	// Struct of the type { "key": "value" }
+	extraVars: [string] | *[]
+
+	// SSH keys of the hosts 
+	sshKeys?: #SSHKeys
+
+	_cmd: {
+		if len(extraVars) == 0 {
+			// No inventory, no extra vars
+			if inventory == _|_ {
+				"ansible-playbook source/\(playbookSource.mainPlaybook) > \(playbookSource.outputPath)/ansible-dagger-log.txt"
+			}
+
+			// Inventory, no extra vars
+			if inventory != _|_ {
+				"ansible-playbook -i inventory/\(inventory.file) source/\(playbookSource.mainPlaybook) > \(playbookSource.outputPath)/ansible-dagger-log.txt"
+			}
+		}
+		if len(extraVars) > 0 {
+			// No inventory, extra vars
+			if inventory == _|_ {
+				"ansible-playbook  --extra-vars \(strings.Join(extraVars, " ")) source/\(playbookSource.mainPlaybook) > \(playbookSource.outputPath)/ansible-dagger-log.txt"
+			}
+
+			// Inventory, extra vars
+			if inventory != _|_ {
+				"ansible-playbook -i inventory/\(inventory.file)  --extra-vars \(strings.Join(extraVars, " ")) source/\(playbookSource.mainPlaybook) > \(playbookSource.outputPath)/ansible-dagger-log.txt"
+			}
+		}
+
+	}
```

Comment:
Nit: It's clearer to use disjunction instead of if statement.
[See docs for concrete example](https://docs.dagger.io/1226/coding-style#favor-disjunctions-over-if-conditions)